tools/incubator.js

```diff
@@ -32,6 +33,21 @@ const Incubator = new Lang.Class({
         return file.get_path();
     },
 
+    _download_file: function (uri) {
+        let session = new Soup.Session();
+        let message = Soup.Message.new('GET', uri);
+        let input_stream = session.send(message, null);
```

Comment:
I think you can accomplish the same thing without Soup. You can just create a GFile for `uri`, `read()` it, and splice it to the temp file's output stream.